core/src/main/java/org/elasticsearch/action/support/ActiveShardCount.java

```diff
@@ -167,12 +167,10 @@ public boolean enoughShardsActive(final ClusterState clusterState, final String
      * Returns true iff the active shard count in the shard routing table is enough
      * to meet the required shard count represented by this instance.
      */
-    public boolean enoughShardsActive(final IndexShardRoutingTable shardRoutingTable, final IndexMetaData indexMetaData) {
-        if (shardRoutingTable.activeShards().size() < resolve(indexMetaData)) {
-            // not enough active shard copies yet
-            return false;
-        }
-        return true;
+    public EvalResult enoughShardsActive(final IndexShardRoutingTable shardRoutingTable, final IndexMetaData indexMetaData) {
```

Comment:
@bleskes the primary motivation for the `resolve` method was to abstract the value within the class.  you are correct in that we could just pass in the number of replicas instead of the entire `IndexMetaData`, but I believe the `resolve` method has utility, otherwise we have to expose a `getValue` method on `ActiveShardCount`, which seems easily avoided if we keep the `resolve` method.  Thoughts?